net/src/eth/mod.rs

```diff
@@ -68,106 +49,57 @@ impl Eth {
     /// # Errors
     ///
     /// This function will return an error if the specified source or dest [Mac] are invalid.
-    pub fn new(source: Mac, destination: Mac, ether_type: EthType) -> Result<Eth, EthError> {
-        let mut header = Eth(Ethernet2Header {
-            source: source.0,
-            destination: destination.0,
+    pub fn new(
+        source: SourceMac,
+        destination: DestinationMac,
+        ether_type: EthType,
+    ) -> Result<Eth, EthError> {
+        Ok(Eth(Ethernet2Header {
+            source: source.inner().0,
+            destination: destination.inner().0,
             ether_type: ether_type.0,
-        });
-        header
-            .set_source_checked(source)
-            .map_err(EthError::InvalidSource)?;
-        header
-            .set_destination(destination)
-            .map_err(EthError::InvalidDestination)?;
-        Ok(header)
+        }))
     }
 
-    /// Get the source [Mac] of the header.
+    /// Get the source [`Mac`] of the header.
     #[must_use]
-    pub fn source(&self) -> Mac {
-        Mac(self.0.source)
+    pub fn source(&self) -> SourceMac {
```

Comment:
Similarly, here, if an Eth can only exist with valid addresses (due to the constructor constraints), why is it needed the unsafe and the _unchecked(). The same applies for the destination() getter. Am I overlooking at something?